lms/services/exceptions.py

```diff
@@ -27,36 +24,26 @@ class LTIOAuthError(LTILaunchVerificationError):
     """Raised when OAuth signature verification of a launch request fails."""
 
 
-class HAPIError(HTTPInternalServerError):  # pylint: disable=too-many-ancestors
+class ExternalRequestError(ServiceError):
     """
-    A problem with an h API request.
-
-    This exception class is raised whenever an h API request times out or when
-    an unsuccessful, invalid or unexpected response is received from the h API.
-
-    An HAPIError is a 500 Internal Server Error response (HTTPInternalServerError
-    subclass) rather than, say, a 502 Bad Gateway because Cloudflare intercepts
-    gateway error responses and replaces our error page with its own error
-    page, and we don't want that.
-
-    Any arguments or keyword arguments other than ``response`` are passed
-    through to HTTPInternalServerError.
+    A problem with a network request to an external service.
 
     :param response: The response from the HTTP request to the h API
     :type response: requests.Response
     """
 
-    def __init__(self, *args, response=None, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, explanation=None, response=None):
+        super().__init__()
+        self.explanation = explanation
         self.response = response
 
     def __str__(self):
         if self.response is None:
-            return super().__str__()
+            return self.explanation
 
-        # Log the details of the h API response. This goes to both Sentry and
-        # the application's logs. It's helpful for debugging to know how h
-        # responded.
+        # Log the details of the response. This goes to both Sentry and the
```

Comment:
This code doesn't log the exception details directly. Does it mean that this method is called when an uncaught exception is logged, in order to get the representation to log?